File: dal-dao-gen/dal_ftp/63/cs/Dao/AccountcheckDao.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using Arch.Data;
using Arch.Data.DbEngine;
using com.ctrip.dal.test.test4.Entity.DataModel;
using com.ctrip.dal.test.test4.Interface.IDao;

namespace com.ctrip.dal.test.test4.Dao
{
   /// <summary>
    /// 更多DAL接口功能，请参阅DAL Confluence，地址：
    /// http://conf.ctripcorp.com/display/SysDev/Dal+Fx+API
    /// </summary>
    public partial class AccountcheckDao : IAccountcheckDao
    {
        readonly BaseDao baseDao = BaseDaoFactory.CreateBaseDao("AccCorpDB_INSERT_1");
        
        /// <summary>
        ///  插入Accountcheck
        /// </summary>
        /// <param name="accountcheck">Accountcheck实体对象</param>
        /// <returns>状态代码</returns>
        public int InsertAccountcheck(Accountcheck accountcheck)
        {
            try
            {
                StatementParameterCollection parameters = new StatementParameterCollection();
                parameters.Add(new StatementParameter{ Name = "@AccCheckID", Direction = ParameterDirection.InputOutput, DbType = DbType.Int32, Value = accountcheck.AccCheckID});
                parameters.Add(new StatementParameter{ Name = "@CorpID", Direction = ParameterDirection.Input, DbType = DbType.AnsiString, Value = accountcheck.CorpID});
                parameters.Add(new StatementParameter{ Name = "@BatchNo", Direction = ParameterDirection.Input, DbType = DbType.AnsiString, Value = accountcheck.BatchNo});
                parameters.Add(new StatementParameter{ Name = "@AccountID", Direction = ParameterDirection.Input, DbType = DbType.Int32, Value = accountcheck.AccountID});
                parameters.Add(new StatementParameter{ Name = "@SubAccountID", Direction = ParameterDirection.Input, DbType = DbType.Int32, Value = accountcheck.SubAccountID});
                parameters.Add(new StatementParameter{ Name = "@BatchStatus", Direction = ParameterDirection.Input, DbType = DbType.AnsiStringFixedLength, Value = accountcheck.BatchStatus});
                parameters.Add(new StatementParameter{ Name = "@AccBalanceID", Direction = ParameterDirection.Input, DbType = DbType.Int32, Value = accountcheck.AccBalanceID});
                parameters.Add(new StatementParameter{ Name = "@AccountType", Direction = ParameterDirection.Input, DbType = DbType.AnsiStringFixedLength, Value = accountcheck.AccountType});
                parameters.Add(new StatementParameter{ Name = "@CheckAccType", Direction = ParameterDirection.Input, DbType = DbType.Int32, Value = accountcheck.CheckAccType});
                parameters.Add(new StatementParameter{ Name = "@Operator", Direction = ParameterDirection.Input, DbType = DbType.AnsiString, Value = accountcheck.Operator});
                parameters.Add(new StatementParameter{ Name = "@ModifyTime", Direction = ParameterDirection.Input, DbType = DbType.DateTime, Value = accountcheck.ModifyTime});
                parameters.Add(new StatementParameter{ Name = "@StartDate", Direction = ParameterDirection.Input, DbType = DbType.AnsiStringFixedLength, Value = accountcheck.StartDate});
                parameters.Add(new StatementParameter{ Name = "@EndDate", Direction = ParameterDirection.Input, DbType = DbType.AnsiStringFixedLength, Value = accountcheck.EndDate});
                parameters.Add(new StatementParameter{ Name = "@FltconMoney", Direction = ParameterDirection.Input, DbType = DbType.Currency, Value = accountcheck.FltconMoney});
                parameters.Add(new StatementParameter{ Name = "@HtlHconMoney", Direction = ParameterDirection.Input, DbType = DbType.Currency, Value = accountcheck.HtlHconMoney});
                parameters.Add(new StatementParameter{ Name = "@HtlXconMoney", Direction = ParameterDirection.Input, DbType = DbType.Currency, Value = accountcheck.HtlXconMoney});
                parameters.Add(new StatementParameter{ Name = "@limited", Direction = ParameterDirection.Input, DbType = DbType.Currency, Value = accountcheck.Limited});
                parameters.Add(new StatementParameter{ Name = "@LimitedTemp", Direction = ParameterDirection.Input, DbType = DbType.Currency, Value = accountcheck.LimitedTemp});
                parameters.Add(new StatementParameter{ Name = "@return",  Direction = ParameterDirection.ReturnValue});

                baseDao.ExecSp("spA__accountcheck_i", parameters);

               accountcheck.AccCheckID = (int)parameters["@AccCheckID"].Value;
                return (int)parameters["@return"].Value;
            }
            catch (Exception ex)
            {
                throw new DalException("调用AccountcheckDao时，访问Insert时出错", ex);
            }
        }
        
        /*由于没有PK，不能生成Update和Delete方法
        /// <summary>
        /// 修改Accountcheck
        /// </summary>
        /// <param name="accountcheck">Accountcheck实体对象</param>
        /// <returns>状态代码</returns>
        public int UpdateAccountcheck(Accountcheck accountcheck)
        {
            //没有Update相关SP,请检查数据库后重新生成。
            return 0;
        }
        
        /// <summary>
        /// 删除Accountcheck
        /// </summary>
        /// <param name="accountcheck">Accountcheck实体对象</param>
        /// <returns>状态代码</returns>
        public int DeleteAccountcheck(Accountcheck accountcheck)
        {
            //没有Delete相关SP,请检查数据库后重新生成。
            return 0;
        }
        
        
        */


        /// <summary>
        /// 获取所有Accountcheck信息
        /// </summary>
        /// <returns>Accountcheck列表</returns>
        public IList<Accountcheck> GetAll()
        {
            try
            {
                return baseDao.GetAll<Accountcheck>();
            }
            catch (Exception ex)
            {
                throw new DalException("调用AccountcheckDao时，访问GetAll时出错", ex);
            }
        }
        
        /// <summary>
        /// 取得总记录数
        /// </summary>
        /// <returns>记录数</returns>
        public long Count()
        {
            try
            {
                String sql = "SELECT count(1) from _accountcheck  with (nolock)  ";

                object obj = baseDao.ExecScalar(sql);
                long ret = Convert.ToInt64(obj);
                return ret;
            }
            catch (Exception ex)
            {
                throw new DalException("调用AccountcheckDao时，访问Count时出错", ex);
            }
        }
        
        /// <summary>
        ///  检索Accountcheck，带翻页
        /// </summary>
        /// <param name="obj">Accountcheck实体对象检索条件</param>
        /// <param name="pagesize">每页记录数</param>
        /// <param name="pageNo">页码</param>
        /// <returns>检索结果</returns>
        public IList<Accountcheck> GetListByPage(Accountcheck obj, int pagesize, int pageNo)
        {
             try
            {
                var dic = new StatementParameterCollection();
                StringBuilder sbSql = new StringBuilder(200);

                
                 //计算ROWNUM
                int fromRownum = (pageNo - 1) * pagesize + 1;
                int endRownum = pagesize * pageNo;
                 sbSql.Append("WITH CTE AS ("); //WITH CTE 开始
                sbSql.Append("select row_number() over(order by AccCheckID desc ) as rownum, ");
                sbSql.Append(@"AccCheckID, CorpID, BatchNo, AccountID, SubAccountID, BatchStatus, AccBalanceID, AccountType, CheckAccType, Operator, ModifyTime, StartDate, EndDate, FltconMoney, HtlHconMoney, HtlXconMoney, limited, LimitedTemp, DataChangeLastTime from _accountcheck (nolock) ");

                //包含查询条件
                //StringBuilder whereCondition = new StringBuilder();
                //if (!string.IsNullOrEmpty(obj.Name))
                //{
                //    //人名
                //    whereCondition.Append("Where Name like @Name ");
                //    dic.AddInParameter("@Name", DbType.String, "%" + obj.Name + "%");
                //}
                //sbSql.Append(whereCondition);

                sbSql.Append(")"); //WITH CTE 结束

                // 用 CTE 完成分页
                sbSql.Append(@"select AccCheckID, CorpID, BatchNo, AccountID, SubAccountID, BatchStatus, AccBalanceID, AccountType, CheckAccType, Operator, ModifyTime, StartDate, EndDate, FltconMoney, HtlHconMoney, HtlXconMoney, limited, LimitedTemp, DataChangeLastTime from CTE Where rownum between @from and @end");
                dic.AddInParameter("@from", DbType.Int32, fromRownum);
                dic.AddInParameter("@end", DbType.Int32, endRownum);
                IList<Accountcheck> list = baseDao.SelectList<Accountcheck>(sbSql.ToString(), dic);

                return list;
            }
            catch (Exception ex)
            {
                throw new DalException("调用AccountcheckDao时，访问GetListByPage时出错", ex);
            }
        }






        
    }
}

File: dal-dao-gen/dal_ftp/63/java/Dao/AccountcheckDao.java
package com.ctrip.dal.test.test4;

import java.math.BigDecimal;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Timestamp;
import java.sql.Types;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

import com.ctrip.platform.dal.dao.DalClient;
import com.ctrip.platform.dal.dao.DalClientFactory;
import com.ctrip.platform.dal.dao.DalHints;
import com.ctrip.platform.dal.dao.DalParser;
import com.ctrip.platform.dal.dao.DalTableDao;
import com.ctrip.platform.dal.dao.KeyHolder;
import com.ctrip.platform.dal.dao.StatementParameters;
import com.ctrip.platform.dal.dao.helper.AbstractDalParser;
import com.ctrip.platform.dal.dao.helper.DalRowMapperExtractor;
import com.ctrip.platform.dal.dao.helper.DalScalarExtractor;

public class AccountcheckDao {
	private static final String DATA_BASE = "AccCorpDB_INSERT_1";
	private static final String COUNT_SQL_PATTERN = "SELECT count(1) from _accountcheck";
	private static final String ALL_SQL_PATTERN = "SELECT * FROM _accountcheck";
	private static final String PAGE_SQL_PATTERN = "WITH CTE AS (select *, row_number() over(order by AccCheckID desc ) as rownum" 
			+" from _accountcheck (nolock)) select * from CTE where rownum between %s and %s";

	private static final String INSERT_SP_NAME = "spA__accountcheck_i";
	private static final String RET_CODE = "retcode";
	private static final String UPDATE_COUNT = "update_count";
	
	private DalParser<Accountcheck> parser = new AccountcheckParser();
	private DalScalarExtractor extractor = new DalScalarExtractor();
	private DalRowMapperExtractor<Accountcheck> rowextractor = null;
	private DalTableDao<Accountcheck> client;
	private DalClient baseClient;

	public AccountcheckDao() {
		this.client = new DalTableDao<Accountcheck>(parser);
		this.rowextractor = new DalRowMapperExtractor<Accountcheck>(parser); 
		this.baseClient = DalClientFactory.getClient(DATA_BASE);
	}

	public Accountcheck queryByPk()
			throws SQLException {
		DalHints hints = new DalHints();
		Accountcheck pk = new Accountcheck();
			

		return client.queryByPk(pk, hints);
	}

	public Accountcheck queryByPk(Accountcheck pk)
			throws SQLException {
		DalHints hints = new DalHints();
		return client.queryByPk(pk, hints);
	}
	
	public long count()  throws SQLException
	{
		StatementParameters parameters = new StatementParameters();
		DalHints hints = new DalHints();
		long result = (Long)this.baseClient.query(COUNT_SQL_PATTERN, parameters, hints, extractor);
		return result;
	}
	
	public List<Accountcheck> queryByPage(Accountcheck pk, int pageSize, int pageNo)  throws SQLException {
		if(pageNo < 1 || pageSize < 1) 
			throw new SQLException("Illigal pagesize or pageNo, pls check");
		
        StatementParameters parameters = new StatementParameters();
		DalHints hints = new DalHints();
		
		String sql = "";
		int fromRownum = (pageNo - 1) * pageSize + 1;
        int endRownum = pageSize * pageNo;
		sql = String.format(PAGE_SQL_PATTERN, fromRownum, endRownum);
		return this.baseClient.query(sql, parameters, hints, rowextractor);
	}
	
	public List<Accountcheck> getAll() throws SQLException
	{
		StatementParameters parameters = new StatementParameters();
		DalHints hints = new DalHints();
		List<Accountcheck> result = null;
		result = this.baseClient.query(ALL_SQL_PATTERN, parameters, hints, rowextractor);
		return result;
	}

	public int insert(Accountcheck daoPojo) throws SQLException {
		StatementParameters parameters = new StatementParameters();
		DalHints hints = new DalHints();
		
		String callSql = prepareSpCall(INSERT_SP_NAME, parameters, parser.getFields(daoPojo));

		parameters.registerInOut("@AccCheckID", Types.INTEGER, daoPojo.getAccCheckID());

		Map<String, ?> results = baseClient.call(callSql, parameters, hints);

		Integer accCheckID = (Integer)parameters.get("@AccCheckID", ParameterDirection.InputOutput).getValue();
		
		return (Integer)results.get(RET_CODE);
	}

	public void delete(Accountcheck...daoPojos) throws SQLException {
		DalHints hints = new DalHints();
		client.delete(hints, daoPojos);
	}

	public void update(Accountcheck...daoPojos) throws SQLException {
		DalHints hints = new DalHints();
		client.update(hints, daoPojos);
	}


	private String prepareSpCall(String SpName, StatementParameters parameters, Map<String, ?> fields) {
		client.addParametersByName(parameters, fields);
		String callSql = client.buildCallSql(SpName, fields.size());
		parameters.setResultsParameter(RET_CODE, extractor);
		parameters.setResultsParameter(UPDATE_COUNT);
		return callSql;
	}
	
	private static class AccountcheckParser extends AbstractDalParser<Accountcheck> {
		public static final String DATABASE_NAME = "AccCorpDB_INSERT_1";
		public static final String TABLE_NAME = "_accountcheck";
		private static final String[] COLUMNS = new String[]{
			"AccCheckID",
			"CorpID",
			"BatchNo",
			"AccountID",
			"SubAccountID",
			"BatchStatus",
			"AccBalanceID",
			"AccountType",
			"CheckAccType",
			"Operator",
			"ModifyTime",
			"StartDate",
			"EndDate",
			"FltconMoney",
			"HtlHconMoney",
			"HtlXconMoney",
			"limited",
			"LimitedTemp",
			"DataChange_LastTime",
		};
		
		private static final String[] PRIMARY_KEYS = new String[]{
		};
		
		private static final int[] COLUMN_TYPES = new int[]{
			Types.INTEGER,
			Types.VARCHAR,
			Types.VARCHAR,
			Types.INTEGER,
			Types.INTEGER,
			Types.CHAR,
			Types.INTEGER,
			Types.CHAR,
			Types.INTEGER,
			Types.VARCHAR,
			Types.TIMESTAMP,
			Types.CHAR,
			Types.CHAR,
			Types.DECIMAL,
			Types.DECIMAL,
			Types.DECIMAL,
			Types.DECIMAL,
			Types.DECIMAL,
			Types.TIMESTAMP,
		};
		
		public AccountcheckParser() {
			super(DATABASE_NAME, TABLE_NAME, COLUMNS, PRIMARY_KEYS, COLUMN_TYPES);
		}
		
		@Override
		public Accountcheck map(ResultSet rs, int rowNum) throws SQLException {
			Accountcheck pojo = new Accountcheck();
			
			pojo.setAccCheckID((Integer)rs.getObject("AccCheckID"));
			pojo.setCorpID((String)rs.getObject("CorpID"));
			pojo.setBatchNo((String)rs.getObject("BatchNo"));
			pojo.setAccountID((Integer)rs.getObject("AccountID"));
			pojo.setSubAccountID((Integer)rs.getObject("SubAccountID"));
			pojo.setBatchStatus((String)rs.getObject("BatchStatus"));
			pojo.setAccBalanceID((Integer)rs.getObject("AccBalanceID"));
			pojo.setAccountType((String)rs.getObject("AccountType"));
			pojo.setCheckAccType((Integer)rs.getObject("CheckAccType"));
			pojo.setOperator((String)rs.getObject("Operator"));
			pojo.setModifyTime((Timestamp)rs.getObject("ModifyTime"));
			pojo.setStartDate((String)rs.getObject("StartDate"));
			pojo.setEndDate((String)rs.getObject("EndDate"));
			pojo.setFltconMoney((BigDecimal)rs.getObject("FltconMoney"));
			pojo.setHtlHconMoney((BigDecimal)rs.getObject("HtlHconMoney"));
			pojo.setHtlXconMoney((BigDecimal)rs.getObject("HtlXconMoney"));
			pojo.setLimited((BigDecimal)rs.getObject("limited"));
			pojo.setLimitedTemp((BigDecimal)rs.getObject("LimitedTemp"));
			pojo.setDatachangeLasttime((Timestamp)rs.getObject("DataChange_LastTime"));
	
			return pojo;
		}
	
		@Override
		public boolean isAutoIncrement() {
			return true;
		}
	
		@Override
		public Number getIdentityValue(Accountcheck pojo) {
			return pojo.getAccCheckID();
		}
	
		@Override
		public Map<String, ?> getPrimaryKeys(Accountcheck pojo) {
			Map<String, Object> primaryKeys = new LinkedHashMap<String, Object>();
			
	
			return primaryKeys;
		}
	
		@Override
		public Map<String, ?> getFields(Accountcheck pojo) {
			Map<String, Object> map = new LinkedHashMap<String, Object>();
			
			map.put("AccCheckID", pojo.getAccCheckID());
			map.put("CorpID", pojo.getCorpID());
			map.put("BatchNo", pojo.getBatchNo());
			map.put("AccountID", pojo.getAccountID());
			map.put("SubAccountID", pojo.getSubAccountID());
			map.put("BatchStatus", pojo.getBatchStatus());
			map.put("AccBalanceID", pojo.getAccBalanceID());
			map.put("AccountType", pojo.getAccountType());
			map.put("CheckAccType", pojo.getCheckAccType());
			map.put("Operator", pojo.getOperator());
			map.put("ModifyTime", pojo.getModifyTime());
			map.put("StartDate", pojo.getStartDate());
			map.put("EndDate", pojo.getEndDate());
			map.put("FltconMoney", pojo.getFltconMoney());
			map.put("HtlHconMoney", pojo.getHtlHconMoney());
			map.put("HtlXconMoney", pojo.getHtlXconMoney());
			map.put("limited", pojo.getLimited());
			map.put("LimitedTemp", pojo.getLimitedTemp());
			map.put("DataChange_LastTime", pojo.getDatachangeLasttime());
	
			return map;
		}
	}
}